### ISSUE TITLE
Setting include relation to `undefined` returns all relations instead of none

### ISSUE THREAD

**Issue by cdubz** [prisma/prisma#25245]
### Bug description

Previous to v5.16.0, when I would run a query like this:

```ts
const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: undefined,
    },
  })
```

I would get `User` records without related `Post` records.

Since v5.16.0, the above example now returns `User` records with all related `Post` records (as if I had set `posts: true`).

### How to reproduce

1. Follow the [Quickstart guide](https://www.prisma.io/docs/getting-started/quickstart).
2. Change the final version of `script.ts` to select all users with `posts: undefined` (instead of `posts: true`).
3. Execute the script -- note that related user posts **are** included.
4. Downgrade the Prisma client: `npm i @prisma/client@5.15.1`
5. Execute the script again -- note that related user posts **are not** included.

### Expected behavior

I have reviewed the releases, issues, and discussion and can't seem to find anything indicating this is an expected change. But, I have had trouble figuring out exactly what to search for.

I am expecting `include: { posts: undefined }` to cause the results to not include any related posts.

### Prisma information

```prisma
model User {
  id    Int     @id @default(autoincrement())
  email String  @unique
  name  String?
  posts Post[]
}

model Post {
  id        Int     @id @default(autoincrement())
  title     String
  content   String?
  published Boolean @default(false)
  author    User    @relation(fields: [authorId], references: [id])
  authorId  Int
}
```

```ts
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

async function main() {
  // After adding a test user and some related posts:
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: undefined,
    },
  })
  console.dir(usersWithPosts, { depth: null })
}

main()
  .then(async () => {
    await prisma.$disconnect()
  })
  .catch(async (e) => {
    console.error(e)
    await prisma.$disconnect()
    process.exit(1)
  })
```

### Environment & setup

- OS: macOS
- Database: SQLite, PostgreSQL
- Node.js version: v20.11.1

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.19.1
@prisma/client          : 5.19.1
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.1
Query Engine (Node-API) : libquery-engine 69d742ee20b815d88e17e54db4a2a7a3b30324e3 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 69d742ee20b815d88e17e54db4a2a7a3b30324e3 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.19.1-2.69d742ee20b815d88e17e54db4a2a7a3b30324e3
Default Engines Hash    : 69d742ee20b815d88e17e54db4a2a7a3b30324e3
Studio                  : 0.502.0
```

**Comment by cdubz** [comment]
I didn't think to test this until just now, but using `include: { posts: false }` gives the desired the result. The change is still unexpected, but certainly easier to deal with that case.

And I also looked at a compare for the version, and am wondering if this change may have been caused by these changes: https://github.com/prisma/prisma/compare/5.15.1...5.16.0#diff-c2ea9dd25b96b17aa7a0263291f1f0260482f388f02188b1d86679b773ce8acfR177-R208

**Comment by jkomyno** [comment]
Hi @cdubz, here's my reproduction attempt.

- `schema.prisma`
  ```prisma
  generator client {
    provider        = "prisma-client-js"
    output          = "../node_modules/.prisma/client"
    previewFeatures = []
  }
  
  datasource db {
    provider = "sqlite"
    url      = "file:./dev.db"
  }
  
  model User {
    id    Int     @id @default(autoincrement())
    name  String  @unique
    posts Post[]
  }
  
  model Post {
    id        Int     @id @default(autoincrement())
    title     String  @unique
    author    User    @relation(fields: [authorId], references: [id])
    authorId  Int
  }
  ```
- `script.ts`, TypeScript snippet
  ```typescript
  import { PrismaClient } from '.prisma/client'
  
  const prisma = new PrismaClient()
  
  async function main() {
    await prisma.post.deleteMany()
    await prisma.user.deleteMany()
  
    await prisma.user.createMany({
      data: [
        {
          id: 1,
          name: 'user-01',
        },
        {
          id: 2,
          name: 'user-02',
        },
        {
          id: 3,
          name: 'user-03',
        },
      ]
    })
  
    await prisma.post.createMany({
      data: [
        {
          id: 1,
          title: 'post-01',
          authorId: 1,
        },
        {
          id: 2,
          title: 'post-02',
          authorId: 1,
        },
        {
          id: 3,
          title: 'post-03',
          authorId: 2,
        },
        {
          id: 4,
          title: 'post-04',
          authorId: 2,
        },
        {
          id: 5,
          title: 'post-05',
          authorId: 3,
        },
        {
          id: 6,
          title: 'post-06',
          authorId: 3,
        }
      ]
    })
  
    const usersWithPosts = ... // different snippet depending on the version
    console.dir(usersWithPosts, { depth: null })
  }
  
  main()
    .then(async () => {
      await prisma.$disconnect()
    })
    .catch(async (e) => {
      console.error(e)
      await prisma.$disconnect()
      process.exit(1)
    })
  ```

## Prisma 5.20.0

### `posts: undefined`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: undefined,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  { id: 1, name: 'user-01' },
  { id: 2, name: 'user-02' },
  { id: 3, name: 'user-03' }
]
```

### `posts: false`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: false,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  { id: 1, name: 'user-01' },
  { id: 2, name: 'user-02' },
  { id: 3, name: 'user-03' }
]
```

### `posts: true`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: true,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  {
    id: 1,
    name: 'user-01',
    posts: [
      { id: 1, title: 'post-01', authorId: 1 },
      { id: 2, title: 'post-02', authorId: 1 }
    ]
  },
  {
    id: 2,
    name: 'user-02',
    posts: [
      { id: 3, title: 'post-03', authorId: 2 },
      { id: 4, title: 'post-04', authorId: 2 }
    ]
  },
  {
    id: 3,
    name: 'user-03',
    posts: [
      { id: 5, title: 'post-05', authorId: 3 },
      { id: 6, title: 'post-06', authorId: 3 }
    ]
  }
]
```

## Prisma 5.16.0

### `posts: undefined`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: undefined,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  {
    id: 1,
    name: 'user-01',
    posts: [
      { id: 1, title: 'post-01', authorId: 1 },
      { id: 2, title: 'post-02', authorId: 1 }
    ]
  },
  {
    id: 2,
    name: 'user-02',
    posts: [
      { id: 3, title: 'post-03', authorId: 2 },
      { id: 4, title: 'post-04', authorId: 2 }
    ]
  },
  {
    id: 3,
    name: 'user-03',
    posts: [
      { id: 5, title: 'post-05', authorId: 3 },
      { id: 6, title: 'post-06', authorId: 3 }
    ]
  }
]
```

### `posts: false`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: false,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  { id: 1, name: 'user-01' },
  { id: 2, name: 'user-02' },
  { id: 3, name: 'user-03' }
]
```

### `posts: true`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: true,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  {
    id: 1,
    name: 'user-01',
    posts: [
      { id: 1, title: 'post-01', authorId: 1 },
      { id: 2, title: 'post-02', authorId: 1 }
    ]
  },
  {
    id: 2,
    name: 'user-02',
    posts: [
      { id: 3, title: 'post-03', authorId: 2 },
      { id: 4, title: 'post-04', authorId: 2 }
    ]
  },
  {
    id: 3,
    name: 'user-03',
    posts: [
      { id: 5, title: 'post-05', authorId: 3 },
      { id: 6, title: 'post-06', authorId: 3 }
    ]
  }
]
```

## Prisma 5.15.1

### `posts: undefined`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: undefined,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  { id: 1, name: 'user-01' },
  { id: 2, name: 'user-02' },
  { id: 3, name: 'user-03' }
]
```

### `posts: false`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: false,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  { id: 1, name: 'user-01' },
  { id: 2, name: 'user-02' },
  { id: 3, name: 'user-03' }
]
```

### `posts: true`

```typescript
  const usersWithPosts = await prisma.user.findMany({
    include: {
      posts: true,
    },
  })
  console.dir(usersWithPosts, { depth: null })
```

Output

```
[
  {
    id: 1,
    name: 'user-01',
    posts: [
      { id: 1, title: 'post-01', authorId: 1 },
      { id: 2, title: 'post-02', authorId: 1 }
    ]
  },
  {
    id: 2,
    name: 'user-02',
    posts: [
      { id: 3, title: 'post-03', authorId: 2 },
      { id: 4, title: 'post-04', authorId: 2 }
    ]
  },
  {
    id: 3,
    name: 'user-03',
    posts: [
      { id: 5, title: 'post-05', authorId: 3 },
      { id: 6, title: 'post-06', authorId: 3 }
    ]
  }
]
```

**Comment by cdubz** [comment]
@jkomyno thanks for taking the time to reproduce (and provide a better script 😁).

I confirmed this is resolved and working as expected in `prisma@5.20.0-dev.21` 🎉

**Comment by jkomyno** [comment]
Amazing! Let's close this issue as solved then!
